subsys/bluetooth/controller/crypto.c

```diff
@@ -36,25 +36,19 @@ int bt_encrypt_le(const uint8_t key[BT_ECB_BLOCK_SIZE],
 		  const uint8_t plaintext[BT_ECB_BLOCK_SIZE],
 		  uint8_t enc_data[BT_ECB_BLOCK_SIZE])
 {
-	uint8_t key_le[BT_ECB_BLOCK_SIZE];
-	uint8_t plaintext_le[BT_ECB_BLOCK_SIZE];
-	uint8_t enc_data_le[BT_ECB_BLOCK_SIZE];
-
 	LOG_HEXDUMP_DBG(key, BT_ECB_BLOCK_SIZE, "key");
 	LOG_HEXDUMP_DBG(plaintext, BT_ECB_BLOCK_SIZE, "plaintext");
 
-	sys_memcpy_swap(key_le, key, BT_ECB_BLOCK_SIZE);
-	sys_memcpy_swap(plaintext_le, plaintext, BT_ECB_BLOCK_SIZE);
-
 	int errcode = MULTITHREADING_LOCK_ACQUIRE();
```

Comment:
do we need this?

Reply:
I think the case that this guards against is the one where this thread and mpsl have different priorities. Without this mutex, if this is running in a preemptive thread for instance, the MPSL thread can preempt it.

I don't think this is different from being preempted by interrupts really. Ideally we would have a test with all three priorities and remove this